core/src/com/unciv/models/ruleset/unit/BaseUnit.kt

```diff
@@ -112,10 +112,18 @@ class BaseUnit : INamed, IConstruction {
 
     override fun getGoldCost(civInfo: CivilizationInfo): Int {
         var cost = getBaseGoldCost(civInfo)
-        if (civInfo.hasUnique("Gold cost of purchasing units -33%")) cost *= 0.66f
+        for (unique in civInfo.getMatchingUniques("Gold cost of purchasing [] units -[]%")) {
+            if (matchesFilter(unique.params[0]))
```

Comment:
Maybe "[] units cost []% less Gold to purchase"

Your thoughts?

Reply:
I don't have a huge preference either way, so I kept it as is.